app/app.go

```diff
@@ -399,6 +414,13 @@ func NewGaiaApp(
 		),
 	)
 	app.SetEndBlocker(app.EndBlocker)
+	app.UpgradeKeeper.SetUpgradeHandler("Gravity-DEX",
+		func(ctx sdk.Context, plan upgradetypes.Plan) {
+			var genState liquiditytypes.GenesisState
+			genState.Params = liquiditytypes.DefaultParams()
+			genState.Params.PoolCreationFee = sdk.NewCoins(sdk.NewCoin("uatom", sdk.NewInt(100000000)))
+			app.LiquidityKeeper.InitGenesis(ctx, genState)
+		})
```

Comment:
```suggestion
		})
		
		upgradeInfo, err := app.UpgradeKeeper.ReadUpgradeInfoFromDisk()
                if err != nil {
                  panic(err)
                 }
                
                if upgradeInfo.Name == "Gravity-DEX" && !app.UpgradeKeeper.IsSkipHeight(upgradeInfo.Height) {
	                storeUpgrades := store.StoreUpgrades{
		                Added: []string{liquiditytypes.ModuleName},
	                }
                
	                // configure store loader that checks if version == upgradeHeight and applies store upgrades
	                app.SetStoreLoader(upgrade.UpgradeStoreLoader(upgradeInfo.Height, &storeUpgrades))
                }
```

Reply:
I applied the code you suggested in the last [commit](https://github.com/cosmos/gaia/pull/859/commits/022f6eea5ec93995bc175c556e37e30cae22e1f6). 
The upgrade was successful based on the cosmosub-4 states with the applied code, and I will organize the details and share them again.
Thanks!